### PR TITLE
Changes Hdfs to use ResultantMonad from omnitool.

### PR DESCRIPTION
**Pull request by stephanh**

### CODE REVIEW

File: src/main/scala/au/com/cba/omnia/permafrost/hdfs/Hdfs.scala [hunk 1]
```diff
@@ -42,202 +43,83 @@ import au.com.cba.omnia.permafrost.io.Streams
...
     Hdfs(c => Result.safe(f(c)))
 
-  /** Build a HDFS operation from a value. The resultant HDFS operation will not throw an exception. */
-  def value[A](v: => A): Hdfs[A] =
-    hdfs(_ => v)
-
   /** Get the HDFS FileSystem for the current configuration. */
   def filesystem: Hdfs[FileSystem] =
-    hdfs(c => FileSystem.get(c))
+    Hdfs.hdfs(c => FileSystem.get(c))
```
Comment (SamRoberts) [inline src/main/scala/au/com/cba/omnia/permafrost/hdfs/Hdfs.scala]: I don't have any particular objections to explicitly putting the object name here. But that's not the way most of our code looks. Any particular reason for the change?

**Comment by SamRoberts** [comment]
:+1:

File: src/main/scala/au/com/cba/omnia/permafrost/hdfs/Hdfs.scala (hunk 1 above)
Comment (stephanh) [inline src/main/scala/au/com/cba/omnia/permafrost/hdfs/Hdfs.scala]: When I was fixing things it was needed for the code to compile. It is no longer needed.

Do you have a preference either way. I don't think we are really consistent across our code base unless you think differently.

I personally like having the `Hdfs.` explicitly.

Comment (SamRoberts) [inline src/main/scala/au/com/cba/omnia/permafrost/hdfs/Hdfs.scala]: Hrmm, off the top of my head I can't think of many times where the objects' methods explicitly use the object's name. I don't mind though: if I was working on a greenfield code base I would probably do the same thing for `Hdfs`. But probably not for something with a much longer object name.

**Pull request merged**